### PR TITLE
Add typehint in geomicric

### PR DESCRIPTION
**Pull request by triple-mu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/image/geometric.py [hunk 1]
```diff
@@ -107,15 +111,17 @@ def imresize(img,
...
-                         keep_ratio=False,
-                         return_scale=False,
-                         interpolation='bilinear',
-                         out=None,
-                         backend=None):
+def imresize_to_multiple(
+    img: np.ndarray,
+    divisor: Union[int, Tuple[int, int]],
+    size: Optional[Union[Tuple[int, int], int]] = None,
+    scale_factor: Optional[Union[Tuple[int, int], int]] = None,
```
Comment (HAOCHENYE) [inline mmcv/image/geometric.py:118]: scale_factor shoule be float

Comment (triple-Mu) [reply mmcv/image/geometric.py]: Fixed

File: mmcv/image/geometric.py [hunk 2]
```diff
@@ -13,7 +14,8 @@
     Image = None
 
 
-def _scale_size(size, scale):
+def _scale_size(size: Tuple[int, int],
+                scale: Union[float, Tuple[float, float]]) -> Tuple[int, int]:
```
Comment (zhouzaida) [inline mmcv/image/geometric.py:18]: scale can also be an integer.

File: mmcv/image/geometric.py [hunk 3]
```diff
@@ -107,15 +111,17 @@ def imresize(img,
...
-                         scale_factor=None,
-                         keep_ratio=False,
-                         return_scale=False,
-                         interpolation='bilinear',
-                         out=None,
-                         backend=None):
+def imresize_to_multiple(
+    img: np.ndarray,
+    divisor: Union[int, Tuple[int, int]],
+    size: Optional[Union[Tuple[int, int], int]] = None,
```
Comment (zhouzaida) [inline mmcv/image/geometric.py:117]:
```suggestion
    size: Union[Tuple[int, int], int, None] = None,
```

File: mmcv/image/geometric.py [hunk 4]
```diff
@@ -107,15 +111,17 @@ def imresize(img,
...
+    scale_factor: Optional[Union[Tuple[float, float], float]] = None,
```
Comment (zhouzaida) [inline mmcv/image/geometric.py:118]:
```suggestion
    scale_factor: Union[Tuple[float, float], float, None] = None,
```

**Pull request closed without merging**